src/sagemaker/jumpstart/artifacts/predictors.py

```diff
@@ -0,0 +1,422 @@
+
+    serializer_type = CONTENT_TYPE_TO_SERIALIZER_TYPE_MAP.get(content_type)
+
+    if serializer_type is None:
+        raise RuntimeError(f"Unrecognized content type: {content_type}")
+
+    serializer_handle = SERIALIZER_TYPE_TO_CLASS_MAP.get(serializer_type)
+
+    if serializer_handle is None:
+        raise RuntimeError(f"Unrecognized serializer type: {serializer_type}")
+
+    return serializer_handle.__call__()
+
+
+def _retrieve_deserializer_from_accept_type(
+    accept_type: MIMEType,
+) -> BaseDeserializer:
+    """Returns deserializer object to use for accept type."""
+
+    deserializer_type = ACCEPT_TYPE_TO_DESERIALIZER_TYPE_MAP.get(accept_type)
+
+    if deserializer_type is None:
+        raise RuntimeError(f"Unrecognized accept type: {accept_type}")
+
+    deserializer_handle = DESERIALIZER_TYPE_TO_CLASS_MAP.get(deserializer_type)
+
+    if deserializer_handle is None:
+        raise RuntimeError(f"Unrecognized deserializer type: {deserializer_type}")
+
+    return deserializer_handle.__call__()
+
+
+def _retrieve_default_deserializer(
+    model_id: str,
+    model_version: str,
+    region: Optional[str],
+    tolerate_vulnerable_model: bool = False,
+    tolerate_deprecated_model: bool = False,
+) -> BaseDeserializer:
+    """Retrieves the default deserializer for the model.
+
+    Args:
+        model_id (str): JumpStart model ID of the JumpStart model for which to
+            retrieve the default deserializer.
+        model_version (str): Version of the JumpStart model for which to retrieve the
+            default deserializer.
+        region (Optional[str]): Region for which to retrieve default deserializer.
+        tolerate_vulnerable_model (bool): True if vulnerable versions of model
+            specifications should be tolerated (exception not raised). If False, raises an
+            exception if the script used by this version of the model has dependencies with known
+            security vulnerabilities. (Default: False).
+        tolerate_deprecated_model (bool): True if deprecated versions of model
+            specifications should be tolerated (exception not raised). If False, raises
+            an exception if the version of the model is deprecated. (Default: False).
+
+    Returns:
+        BaseDeserializer: the default deserializer to use for the model.
+    """
+
+    default_accept_type = _retrieve_default_accept_type(
+        model_id=model_id,
+        model_version=model_version,
+        region=region,
+        tolerate_vulnerable_model=tolerate_vulnerable_model,
+        tolerate_deprecated_model=tolerate_deprecated_model,
+    )
+
+    return _retrieve_deserializer_from_accept_type(MIMEType.from_suffixed_type(default_accept_type))
+
+
+def _retrieve_default_serializer(
+    model_id: str,
+    model_version: str,
+    region: Optional[str],
+    tolerate_vulnerable_model: bool = False,
+    tolerate_deprecated_model: bool = False,
+) -> BaseSerializer:
```

Comment:
sanity-check: Does it return an (a) instance of a child class of `BaseSerializer`, or (b) the child class itself?

I _think_ it is (a), in which case your typing is correct, just want to double check.

Reply:
It is (a)